pgjdbc/src/main/java/org/postgresql/jdbcurlresolver/PgPassParser.java

```diff
@@ -67,84 +69,50 @@ private PgPassParser(String hostname, String port, String database, String user)
-    } catch ( MalformedURLException ex ) {
-      // try file
-      File file = new File(resourceName);
-      return new FileInputStream(file);
-    }
+  // open File
+  private @Nullable InputStream openInputStream(String resourceName) throws IOException {
+    Path path = new File(resourceName).toPath();
+    return checkFilePermissions(path) ? Files.newInputStream(path) : null;
   }
 
-  // choose resource where to search for service description
-  private @Nullable String findPgPasswordResourceName() {
-    // default file name
-    String pgPassFileDefaultName = PGEnvironment.PGPASSFILE.getDefaultValue();
-
-    // if there is value, use it - 1st priority
-    {
-      String propertyName = PGEnvironment.ORG_POSTGRESQL_PGPASSFILE.getName();
-      String resourceName = System.getProperty(propertyName);
-      if (resourceName != null && !resourceName.trim().isEmpty()) {
-        LOGGER.log(Level.FINE, "Value [{0}] selected from property [{1}]", new Object[]{resourceName, propertyName});
-        return resourceName;
-      }
-    }
-
-    // if there is value, use it - 2nd priority
-    {
-      String envVariableName = PGEnvironment.PGPASSFILE.getName();
-      String resourceName = System.getenv().get(envVariableName);
-      if (resourceName != null && !resourceName.trim().isEmpty()) {
-        LOGGER.log(Level.FINE, "Value [{0}] selected from environment variable [{1}]", new Object[]{resourceName, envVariableName});
-        return resourceName;
-      }
-    }
-
-    // if file in user home is readable, use it, otherwise continue - 3rd priority
-    {
-      String resourceName = "";
-      if ( !OSUtil.isWindows() ) {
-        resourceName += ".";
-      }
-      resourceName += pgPassFileDefaultName;
-      if (OSUtil.isWindows()) {
-        resourceName += ".conf";
-      }
-      File resourceFile = new File(OSUtil.getUserConfigRootDirectory(), resourceName);
-      if (resourceFile.canRead()) {
-        LOGGER.log(Level.FINE, "Value [{0}] selected because file exist in user home directory", new Object[]{resourceFile.getAbsolutePath()});
-        return resourceFile.getAbsolutePath();
+  // in case there are permissions for "group" or "other" then return false
+  private boolean checkFilePermissions(Path path) throws IOException {
+    FileSystem fileSystem = path.getFileSystem();
+    // check works for posix filesystems
+    if (fileSystem.supportedFileAttributeViews().contains("posix")) {
+      Set<PosixFilePermission> posixFilePermissions = Files.getPosixFilePermissions(path);
+      for (PosixFilePermission filePermission : posixFilePermissions) {
+        switch (filePermission) {
+          case GROUP_READ:
+          case GROUP_WRITE:
+          case GROUP_EXECUTE:
+          case OTHERS_READ:
+          case OTHERS_WRITE:
+          case OTHERS_EXECUTE:
+            LOGGER.log(Level.WARNING, "password file [{0}] has group or world access; permissions should be u=rw (0600) or less", new Object[]{fileName});
```

Comment:
I suggest we log the current value along with the desired one

Reply:
current value is logged now: https://github.com/pgjdbc/pgjdbc/pull/2646/commits/a81210d89dfe20c5c2ed22976639ea7e306c4a63